lnp2p/peer.go

```diff
@@ -8,66 +8,20 @@ import (
 
 // A Peer is a remote client that's somehow connected to us.
 type Peer struct {
-	lnaddr   lncore.LnAddr
-	nickname *string
-	conn     *lndc.Conn
-	idpubkey pubkey
+	Addr     string
+	Idx      uint32 // deprecated, default value for all uint32s are 0
+	Nickname string
```

Comment:
I don't like having these things exported since it makes it much more difficult to change the implementation later.

Reply:
Yeah, I thought about this for a while as well, but we have function maps like `instance.GetVal()` for things which should be as easy as `instance.Val`. If we're using the accessor function everywhere like we do right now, we might as well export the value itself.